tests/virt_autotest/update_package.pm

```diff
@@ -62,6 +62,11 @@ sub run {
         assert_script_run("sed -irn \"s/console=ttyAMA0/console=$ipmi_console/g\" /usr/share/qa/virtautolib/lib/vh-update-lib.sh");
     }
 
+
+    #generate ssh key on host
+    #TODO: need on sles11sp4 only??? move from sr-iov here as a general step. Need remove it from virtual network test?
+    script_run "rm ~/.ssh/id_rsa*; ssh-keygen -t rsa -f ~/.ssh/id_rsa -P ''";
```

Comment:
Why do you removing old keys and generating new?
```suggestion
    script_run "ssh-keygen -t rsa -f ~/.ssh/id_rsa -P ''";
```

Reply:
@guoxuguang  suggest remove and generate a new key because some host with older release may have some sshkey problem.  He met this issue in his virtual network tests. @guoxuguang , can you help give some input here?

Reply:
sure, imho, there were the following questions about ssh key:
1. Why remove the old keys and created new ssh keys?
As my understood, there was a known product about ssh key on sles12sp5 or sles11sp4 vm host, need to
remove the old keys and create new ssh keys as recommente
 2. Why created ssh key on sles vm host
imho, if there was not created ssh key (i.e. RAS ssh key)  as default, we'd better to create new ssh keyRSA KEY) on sles vm host  for the follow virt_auto test as required. 
Feel free to let me know if there was any question about ssh key here.

Reply:
@pdostal As Leon met problem in some product release, we can avoid it by just removing the existing keys. Our test focus is on virtualization, and we don't care much about sshkeys, as long as we can run test and try to avoid other aspects of blocking test. Do you agree?

Reply:
@Julie-CAO not really. Our tests are queued so if you change the SSH key at the middle of the testing process everything after that will fail.

Reply:
@pdostal 
 - Just did some tests on sles15sp2, 'rm ~/.ssh/id_rsa*; ssh-keygen XXX' does change the SSH key; 'ssh-keygen XXX' only without 'rm rsa.*' does not change the key, but it needs interactive input to confirm 'override', so automation test does not work this way.
 - the script to generate a new SSH key is in update_pacakge.pm, which runs just after host installation in our virtualization tests. ie. No actual tests have been run in this step, so the change will not block any subsequent tests. But I know little about QAM tests, does your tests call update_package.pm this way?
 - Even though the change will not cause any trouble to subsequent tests, it is better to be improved. I can change to '[ -f rsa.pub ] && re-generate sshkey', then the SSH key will not change.
 - @guoxuguang, the virtual network tests seem to having SSH key changing problem mentioned by Pavel. You need make change to avoid of blocking other tests.

Reply:
> @pdostal
> 
>     * Just did some tests on sles15sp2, 'rm ~/.ssh/id_rsa*; ssh-keygen XXX' does change the SSH key; 'ssh-keygen XXX' only without 'rm rsa.*' does not change the key, but it needs interactive input to confirm 'override', so automation test does not work this way.
> 
>     * the script to generate a new SSH key is in update_pacakge.pm, which runs just after host installation in our virtualization tests. ie. No actual tests have been run in this step, so the change will not block any subsequent tests. But I know little about QAM tests, does your tests call update_package.pm this way?
> 
>     * Even though the change will not cause any trouble to subsequent tests, it is better to be improved. I can change to '[ -f rsa.pub ] && re-generate sshkey', then the SSH key will not change.
> 
>     * @guoxuguang, the virtual network tests seem to having SSH key changing problem mentioned by Pavel. You need make change to avoid of blocking other tests.

sure, got it. 
There should be created a new PR to fix this SSH key changing problem  as your mentioned.

Reply:
Ad) Just did some tests on sles15sp2, 'rm ~/.ssh/id_rsa*; ssh-keygen XXX' does change the SSH key; 'ssh-keygen XXX' only without 'rm rsa.*' does not change the key, but it needs interactive input to confirm 'override', so automation test does not work this way.

@Julie-CAO This is not true.
```bash
pdostal-server:~$ md5sum ~/.ssh/id_rsa*
cf94f289716eb7b42719d31b85602b77  /home/pdostal/.ssh/id_rsa
dfdc5d028e85e5d3f69c62ec77204f0c  /home/pdostal/.ssh/id_rsa.pub
pdostal-server:~$ ssh-keygen -t rsa
Generating public/private rsa key pair.
Enter file in which to save the key (/home/pdostal/.ssh/id_rsa):
/home/pdostal/.ssh/id_rsa already exists.
Overwrite (y/n)? y
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Your identification has been saved in /home/pdostal/.ssh/id_rsa.
Your public key has been saved in /home/pdostal/.ssh/id_rsa.pub.
The key fingerprint is:
SHA256:184n447dn2L1fCsT44iamNrcAc7miFmfiLSkpBL+R0w pdostal@pdostal-server
pdostal-server:~$ md5sum ~/.ssh/id_rsa*
d0e9c5b0d97b527cfd2ecbbcf46fb71e  /home/pdostal/.ssh/id_rsa
f82cb2e4646c4136637c4c1a48b30232  /home/pdostal/.ssh/id_rsa.pub
```

Reply:
Yes, you are right. I did test again. I might skip overriding original keyfiles in my previous check. Thanks, Pavel.

Reply:
> > @pdostal
> > ```
> > * Just did some tests on sles15sp2, 'rm ~/.ssh/id_rsa*; ssh-keygen XXX' does change the SSH key; 'ssh-keygen XXX' only without 'rm rsa.*' does not change the key, but it needs interactive input to confirm 'override', so automation test does not work this way.
> > 
> > * the script to generate a new SSH key is in update_pacakge.pm, which runs just after host installation in our virtualization tests. ie. No actual tests have been run in this step, so the change will not block any subsequent tests. But I know little about QAM tests, does your tests call update_package.pm this way?
> > 
> > * Even though the change will not cause any trouble to subsequent tests, it is better to be improved. I can change to '[ -f rsa.pub ] && re-generate sshkey', then the SSH key will not change.
> > 
> > * @guoxuguang, the virtual network tests seem to having SSH key changing problem mentioned by Pavel. You need make change to avoid of blocking other tests.
> > ```
> 
> sure, got it.
> There should be created a new PR to fix this SSH key changing problem as your mentioned.

Quick update, PR#10555, was included this SSH key changing problem. 
Feel free to follow PR#10555 for more details:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10555